client/app/nasha/partition/snapshot/nasha-partition-snapshot.html

```diff
@@ -1,78 +1,44 @@
-               data-tooltip-placement="right"></i>
-        </p>
-
-        <div class="m-3">
-            <table class="table">
-                <tbody data-ng-show="NashaPartitionSnapshotCtrl.loading">
-                    <tr>
-                        <td class="text-center">
-                            <oui-spinner></oui-spinner>
-                        </td>
-                    </tr>
-                </tbody>
-
-                <tbody data-ng-hide="NashaPartitionSnapshotCtrl.loading">
-                    <tr data-ng-repeat="snapshot in NashaPartitionSnapshotCtrl.snapshotEnum">
-                        <td class="py-3">
-                            <div class="oui-checkbox m-0">
-                                <input class="oui-checkbox__input"
-                                    id="{{ 'frequency_' + $index }}"
-                                    name="{{ 'frequency_' + $index }}"
-                                    type="checkbox"
-                                    data-ng-checked="NashaPartitionSnapshotCtrl.isScheduled(snapshot)"
-                                    data-ng-click="NashaPartitionSnapshotCtrl.changeSchedule(snapshot, !NashaPartitionSnapshotCtrl.isScheduled(snapshot))">
-                                <label class="oui-checkbox__label-container"
-                                    for="{{ 'frequency_' + $index }}">
-                                    <span class="oui-checkbox__label m-0" >
-                                        <span data-translate="{{'nasha_snapshot_'+ snapshot}}"></span>
-                                        <span class="oui-checkbox__icon">
-                                            <i class="oui-icon oui-icon-checkbox-unchecked" aria-hidden="true"></i>
-                                            <i class="oui-icon oui-icon-checkbox-checked" aria-hidden="true"></i>
-                                            <i class="oui-icon oui-icon-checkbox-checkmark" aria-hidden="true"></i>
-                                        </span>
-                                    </span>
-                                </label>
-                            </div>
-                        </td>
-                    </tr>
-                </tbody>
-            </table>
+    <oui-field data-label="{{ :: 'nasha_snapshot_frequency' | translate }}"
+        data-label-popover="{{ :: 'nasha_snapshot_informations' | translate }}">
+        <div class="oui-checkbox"
+             data-ng-repeat="snapshot in NashaPartitionSnapshotCtrl.snapshotEnum">
+            <input class="oui-checkbox__input"
+                id="{{ 'frequency_' + $index }}"
+                name="{{ 'frequency_' + $index }}"
+                type="checkbox"
+                data-ng-checked="NashaPartitionSnapshotCtrl.isScheduled(snapshot)"
+                data-ng-click="NashaPartitionSnapshotCtrl.changeSchedule(snapshot, !NashaPartitionSnapshotCtrl.isScheduled(snapshot))">
```

Comment:
the ngClick is needed ?

I think you should use a checkbox + ng-model, the model will be set automatically, no? (If you need custom value, you can use ngTrueValue and ngFalseValue.)

Reply:
What if he needs custom behavior when the value changes ?

Reply:
You should use ngChange event :)

Reply:
I just realized it's an `input`, of course you're right

Reply:
haven't done that part of code but ngChange would require a ngModel. And I've only got ngChecked there following the way of it has been constructed.